internal/pkg/prune/prune.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prune
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/cli-experimental/internal/pkg/client"
+	"sigs.k8s.io/cli-experimental/internal/pkg/clik8s"
+	"sigs.k8s.io/kustomize/pkg/inventory"
+)
+
+// Prune prunes directories
+type Prune struct {
+	DynamicClient client.Client
+	Out           io.Writer
+	Resources     clik8s.ResourcePruneConfigs
+	Commit        *object.Commit
+}
+
+// Result contains the Prune Result
+type Result struct {
+	Resources clik8s.ResourceConfigs
+}
+
+// Do executes the prune
```

Comment:
It may be worthy to provide high-level details of the pruning operation, or a link to an open-sourced document should be sufficient.